Tools/cases_generator/analyzer.py

```diff
@@ -347,10 +347,25 @@ def desugar_inst(
 ) -> None:
     assert inst.kind == "inst"
     name = inst.name
-    uop = make_uop("_" + inst.name, inst)
+    op_inputs: list[parser.InputEffect] = []
+    parts: list[Part] = []
+    uop_index = -1
+    for input in inst.inputs:
+        if isinstance(input, parser.CacheEffect) and input.name == "unused":
+            parts.append(Skip(input.size))
+        else:
+            op_inputs.append(input)
+            if uop_index < 0:
+                uop_index = len(parts)
+                parts.append(Skip(0))
+    uop = make_uop("_" + inst.name, inst, op_inputs)
     uop.implicitly_created = True
     uops[inst.name] = uop
-    add_instruction(name, [uop], instructions)
+    if uop_index < 0:
+        parts.append(uop)
+    else:
+        parts[uop_index] = uop
+    add_instruction(name, parts, instructions)
```

Comment:
Something here feels so clever it needs a comment. and/or an assert We're adding a `Skip(0)` in L360 and overwriting that in L367. Could we ever add multiple `Skip(0)` entries? Then we'd be overwriting the first `Skip(0)` only. Probably my question makes no sense, and a comment would clarify that.

Separately, maybe `Skip` should have been named `Unused`? Or even `UnusedCacheEffect`? Because that's what it represents, right?

Reply:
We have to add `Skip(0)` due to type constraints. The natural thing to add would be `None`, but the then we need to add casts, or messy types.

`Skip` skips over caches entries, so I think the name is better than `Unused`. We could add `Cache` though.
A bit out of scope for this PR, though.